--- libdrm-1.0.3/shared-core/via_drm.h.orig	2005-08-15 18:07:12.000000000 +0000
+++ libdrm-1.0.3/shared-core/via_drm.h	2005-10-28 07:53:13.000000000 +0000
@@ -103,8 +103,13 @@
 #define VIA_BACK    0x2
 #define VIA_DEPTH   0x4
 #define VIA_STENCIL 0x8
-#define VIDEO 0
-#define AGP 1
+
+#define VIA_MEM_VIDEO	0	/* matches drm constant */
+#define VIA_MEM_AGP	1	/* matches drm constant */
+#define VIA_MEM_SYSTEM	2
+#define VIA_MEM_MIXED	3
+#define VIA_MEM_UNKNOWN	4
+
 typedef struct {
 	uint32_t offset;
 	uint32_t size;
@@ -192,6 +197,10 @@
 	unsigned int XvMCSubPicOn[VIA_NR_XVMC_PORTS];
 	unsigned int XvMCCtxNoGrabbed;	/* Last context to hold decoder */
 
+	/* Used by the 3d driver only at this point, for pageflipping:
+	 */
+
+	unsigned int pfCurrentOffset;
 } drm_via_sarea_t;
 
 typedef struct _drm_via_cmdbuf_size {
