--- libdrm-2.4.117/meson.build.orig	2023-10-20 11:20:00.675934935 +0200
+++ libdrm-2.4.117/meson.build	2023-10-20 18:29:31.362989811 +0200
@@ -87,7 +87,7 @@ dep_pciaccess = dependency('pciaccess',
 with_intel = get_option('intel') \
   .require(with_atomics, error_message : 'libdrm_intel requires atomics') \
   .require(dep_pciaccess.found(), error_message : 'libdrm_intel requires libpciaccess') \
-  .disable_auto_if(not host_machine.system().startswith('x86')) \
+  .disable_auto_if(not host_machine.cpu_family().startswith('x86')) \
   .allowed()
 summary('Intel', with_intel)
 
