--- libdrm-2.4.56/intel/intel_bufmgr_gem.c~	2014-02-07 17:58:41.000000000 +0100
+++ libdrm-2.4.56/intel/intel_bufmgr_gem.c	2014-09-04 10:14:00.629857376 +0200
@@ -2924,7 +2924,7 @@
 get_pci_device_id(drm_intel_bufmgr_gem *bufmgr_gem)
 {
 	char *devid_override;
-	int devid;
+	int devid = 0;
 	int ret;
 	drm_i915_getparam_t gp;
 
@@ -2936,7 +2936,6 @@
 		}
 	}
 
-	VG_CLEAR(devid);
 	VG_CLEAR(gp);
 	gp.param = I915_PARAM_CHIPSET_ID;
 	gp.value = &devid;
